google_guest_agent/addresses.go

```diff
@@ -532,8 +532,22 @@ func enableNetworkInterfaces() error {
 }
 
 func isNetworkManagerRunning() bool {
-	res := runCmd(exec.Command("systemctl", "is-active", "--quiet", "NetworkManager"))
-	return res == nil
+	// SystemD
+	if systemctl, err := exec.LookPath("systemctl"); err == nil {
+		return runCmd(exec.Command(systemctl, "is-active", "--quiet", "NetworkManager")) == nil
+	}
+	// Upstart
+	if runCmd(exec.Command("/sbin/init", "--version")) == nil {
```

Comment:
all init systems provide an /sbin/init, but i don't know if all of them are safe to invoke directly like this on a running system

Reply:
The test condition I found is like this:
[[ `/sbin/init --version` =~ upstart ]] && echo yes || echo no
Since It's after testing in systemd, so I omit this string match logic.